### PR TITLE
Hotfix/some bugs

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Fixes crash loop with workOrders
- Let's players attack mobs
- Fix crash with not existing file paths.

Review please

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ColonyPermissionEventHandler.java#L31](https://github.com/Minecolonies/minecolonies/blob/4cd8cb02a33f14a4034b60e2c3dc27b177bc7ed6/src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java#L31): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java [hunk 1]
```diff
@@ -373,7 +374,8 @@ public void on(final ArrowLooseEvent event)
     public void on(final AttackEntityEvent event)
     {
         final EntityPlayer playerIn = event.getEntityPlayer();
-        if (Configurations.enableColonyProtection && colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        if (Configurations.enableColonyProtection && colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition())
+                && !(event.getEntity() instanceof EntityMob))
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java]: Move this check to be second, so we don't have to do the isCoordInColonyCheck for mobs

File: src/main/java/com/minecolonies/structures/helpers/Structure.java [hunk 2]
```diff
@@ -105,7 +105,14 @@ public Structure(@Nullable final World world, final String structureName, final
...
                     decorationFolder = new File(FMLCommonHandler.instance().getMinecraftServerInstance().getDataDirectory(), "minecolonies/");
                 }
-                inputstream = new FileInputStream(decorationFolder.getPath() + "/" + structureName + ".nbt");
+                if(decorationFolder.exists())
+                {
+                    inputstream = new FileInputStream(decorationFolder.getPath() + "/" + structureName + ".nbt");
+                }
+                else
+                {
+                    throw new FileNotFoundException();
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/structures/helpers/Structure.java]: Give a message saying the decoration folder doesn't exist.

Comment (Raycoms) [reply src/main/java/com/minecolonies/structures/helpers/Structure.java]: Will do on FileNotFound:
-  Log.getLogger().warn(String.format("Failed to load template %s", structureName), e);

Usually this only happens when the player or server owner removes a schematic a worker is trying to build.

Comment (cltnschlosser) [reply src/main/java/com/minecolonies/structures/helpers/Structure.java]: Still adding a message here will give more info, and it's a simple add

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ColonyPermissionEventHandler.java#L31](https://github.com/Minecolonies/minecolonies/blob/41692df9b3692329652b4863a25b12a13888880b/src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java#L31): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ColonyPermissionEventHandler.java#L31](https://github.com/Minecolonies/minecolonies/blob/2033036b9f436bfa5340e182ef51bff401f33bc8/src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java#L31): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**